.github/workflows/lint.yaml

```
@@ -37,10 +37,14 @@ jobs:
           path: ~/.cache/pip
           key: pip-cache-${{ matrix.python-version }}-${{ hashFiles('lib/galaxy/dependencies/pinned-lint-requirements.txt') }}
       - name: Cache tox env
+        env:
+          LINT_PATH: 'lib/galaxy/dependencies/pinned-lint-requirements.txt'
+          TYPE_PATH: 'lib/galaxy/dependencies/pinned-typecheck-requirements.txt'
+          CORE_PATH: 'lib/galaxy/dependencies/pinned-requirements.txt'
         uses: actions/cache@v3
         with:
           path: .tox
-          key: tox-cache-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('lib/galaxy/dependencies/pinned-lint-requirements.txt') }}-lint
+          key: tox-cache-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles(env.LINT_PATH, env.TYPE_PATH, env.CORE_PATH) }}-lint-and-typecheck
```

I'd keep the key suffix the same as the workflow filename

```suggestion
          key: tox-cache-${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles(env.LINT_PATH, env.TYPE_PATH, env.CORE_PATH) }}-lint
```